test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java

```diff
@@ -0,0 +1,272 @@
+        aCompoundCommand1.add(aAddEdgeCommand);
+        aCompoundCommand1.add(aMoveCommand1);
+        
+        aCompoundCommand2 = new CompoundCommand();
+        aCompoundCommand2.add(aAddActorNodeCommand);
+        aCompoundCommand2.add(aMoveCommand2);
+        
+        aCompoundCommand3 = new CompoundCommand();
+        aCompoundCommand3.add(aCompoundCommand1);
+        aCompoundCommand3.add(aCompoundCommand2);
+        
+        aCompoundCommand4 = new CompoundCommand();
+        aCompoundCommand4.add(aCompoundCommand1);
+        aCompoundCommand4.add(aMoveCommand2);
+        aCompoundCommand4.add(aDeletNodeCommand);
+    }
+
+    @Test
+    public void testExecute_ContainsSimpleCommands() 
+    {
+    	assertFalse(aGraph.getRootNodes().contains(aClassNode));
+        assertFalse(aGraph.getEdges().contains(aCallEdge));
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(0, 0, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
+       
+        aCompoundCommand1.execute();        
+        
+        assertTrue(aGraph.getRootNodes().contains(aClassNode));
+        assertTrue(aGraph.getEdges().contains(aCallEdge));
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(5, 5, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
+    }
+    
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testUndo_ContainsSimpleCommands() 
+    {
+        aCompoundCommand1.execute();
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(5, 5, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
+        try 
+        {
+            ArrayList<Node> aListNodesToBeRemoved = (ArrayList<Node>) (aNodesToBeRemoved.get(aGraph));
+            assertFalse(aListNodesToBeRemoved.contains((Node) aClassNode));
+            ArrayList<Edge> aListEdgesToBeRemoved = (ArrayList<Edge>) aEdgesToBeRemoved.get(aGraph);
+            assertFalse(aListEdgesToBeRemoved.contains(aCallEdge));
+        } 
+        catch (IllegalArgumentException | IllegalAccessException e1) 
+        {
+            fail();
+        }
+        
+        aCompoundCommand1.undo();
+        
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(0, 0, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
+        try 
+        {
+            ArrayList<Node> aListNodesToBeRemoved = (ArrayList<Node>) (aNodesToBeRemoved.get(aGraph));
+            assertTrue(aListNodesToBeRemoved.contains((Node) aClassNode));
+            ArrayList<Edge> aListEdgesToBeRemoved = (ArrayList<Edge>) aEdgesToBeRemoved.get(aGraph);
+            assertTrue(aListEdgesToBeRemoved.contains(aCallEdge));
+        } 
+        catch (IllegalArgumentException | IllegalAccessException e1) 
+        {
+            fail();
+        }
+    }
+    
+    @Test
+    public void testExecute_ContainsCompoundCommands() 
+    {
+    	assertFalse(aGraph.getRootNodes().contains(aClassNode));
```

Comment:
Same comment, the `assertFalse` predicates can be removed, and assumed. Given the `setup` method, you can start the test directly at line 143 (the command execution).